src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -62,110 +132,151 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
 
             this._deferredRequestNonce = { resolve, reject };
             this._paymentForm.requestCardNonce();
-        })
-        .then(paymentData => {
-            const paymentPayload = {
-                methodId: paymentName,
-                paymentData,
-            };
-
-            return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
-                .then(() =>
-                    this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload))
-                );
         });
     }
 
     private _getFormOptions(options: PaymentInitializeOptions, deferred: DeferredPromise): SquareFormOptions {
-        const { square: squareOptions, methodId } = options;
-        const state = this._store.getState();
-        const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+        const { square: squareOptions } = options;
 
-        if (!squareOptions || !paymentMethod) {
+        if (!squareOptions || !this._paymentMethod) {
             throw new InvalidArgumentError('Unable to proceed because "options.square" argument is not provided.');
         }
 
+        this._squareOptions = squareOptions;
+
         return {
-            ...squareOptions,
-            ...paymentMethod.initializationData,
+            ...this._squareOptions,
+            ...this._paymentMethod.initializationData,
             callbacks: {
-                paymentFormLoaded: () => {
-                    deferred.resolve();
-
-                    const state = this._store.getState();
-                    const billingAddress = state.billingAddress.getBillingAddress();
-
-                    if (!this._paymentForm) {
-                        throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
-                    }
-
-                    if (billingAddress && billingAddress.postalCode) {
-                        this._paymentForm.setPostalCode(billingAddress.postalCode);
+                cardNonceResponseReceived: (errors, nonce, cardData, billingContact, shippingContact) => {
+                    if (cardData && cardData.digital_wallet_type !== DigitalWalletType.none) {
+                        this._handleCardNonceResponse(errors, nonce, cardData, billingContact, shippingContact);
```

Comment:
I'm really sorry, I got the method names wrong way around. The current `_handleCardNonceResponse` should be renamed as `_handleWalletNonceResponse` (because it's for handling wallet response), and `_handleWalletNonceResponse` should be renamed as `_handleCardNonceResponse` (because it's for handling the normal CC flow).